water/reverse.go

```diff
@@ -0,0 +1,94 @@
+package water
+
+import (
+	"context"
+	"encoding/base64"
+	"net"
+	"sync"
+
+	"github.com/getlantern/golog"
+	"github.com/refraction-networking/water"
+	_ "github.com/refraction-networking/water/transport/v0"
+)
+
+var log = golog.LoggerFor("water")
+
+type llistener struct {
+	net.Listener
+
+	connections  chan net.Conn
+	closedSignal chan struct{}
+	closeOnce    sync.Once
+	closeError   error
+}
+
+func NewReverseListener(ctx context.Context, address string, wasm string) (*llistener, error) {
+	decodedWASM, err := base64.StdEncoding.DecodeString(wasm)
+	if err != nil {
+		log.Errorf("failed to decode WASM base64: %v", err)
+		return nil, err
+	}
+
+	cfg := &water.Config{
+		TransportModuleBin: decodedWASM,
+	}
+
+	waterListener, err := cfg.ListenContext(ctx, "tcp", address)
```

Comment:
Why can't you just use this listener?  Why all the channels tracking connections and closing?  That should be handled by `waterListener` already, right?

Reply:
I've added to follow the same standard as other implementations, I'll remove that!